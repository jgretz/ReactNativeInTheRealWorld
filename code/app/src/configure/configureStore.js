import {createStore, compose, applyMiddleware} from 'redux';
import thunkMiddleware from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import {composeWithDevTools} from 'remote-redux-devtools';
import {autoRehydrate, persistStore} from 'redux-persist';
import {AsyncStorage} from 'react-native';

import rootReducer from '../rootReducer';
import {storePersistor} from '../util';

const createDevStore = middleware => {
  const devCompose = composeWithDevTools({realtime: true});
  const store = createStore(rootReducer, devCompose(
    applyMiddleware(...middleware),
    autoRehydrate(),
  ));

  if (module.hot) {
    // Enable hot module replacement for reducers
    module.hot.accept('../rootReducer', () => {
      const nextReducer = require('../rootReducer').default; // eslint-disable-line global-require
      store.replaceReducer(nextReducer);
    });
  }

  return store;
};

const createProdStore = middleware =>
  createStore(rootReducer, null, compose(
    applyMiddleware(...middleware),
    autoRehydrate()
  ));

export const configureStore = () => {
  const middleware = [
    thunkMiddleware,
    promiseMiddleware(),
  ];

  const store = __DEV__ ? createDevStore(middleware) : createProdStore(middleware); // eslint-disable-line

  storePersistor(
    persistStore(store, {storage: AsyncStorage})
  );

  return store;
};
