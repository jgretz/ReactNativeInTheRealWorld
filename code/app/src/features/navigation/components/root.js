import React, {Component} from 'react';
import {connect} from 'react-redux';
import {StackNavigator, addNavigationHelpers} from 'react-navigation';

import routes from '../util/routes';
import {navigationSelector} from '../selectors';

// singleton navigator
export const Navigator = StackNavigator(routes,  {headerMode: 'screen'}); // eslint-disable-line

class Root extends Component {
  render() {
    const {dispatch, navigation} = this.props;
    const nav = addNavigationHelpers({dispatch, state: navigation});

    return (
      <Navigator navigation={nav} />
    );
  }
}

const mapStateToProps = state => ({
  navigation: navigationSelector(state),
});

export default connect(mapStateToProps)(Root);
