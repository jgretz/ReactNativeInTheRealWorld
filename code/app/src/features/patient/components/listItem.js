import React, {Component} from 'react';
import {View, Text, TouchableHighlight} from 'react-native';
import {connect} from 'react-redux';
import autobind from 'class-autobind';

import containerStyles from '../../../styles/container';
import listItemStyles from '../styles/listItem';

import {push, PRESCRIPTION_LIST_FOR_PATIENT} from '../../navigation';

// component
class ListItem extends Component {
  constructor(props) {
    super(props);
    autobind(this);
  }

  handlePress() {
    this.props.push(PRESCRIPTION_LIST_FOR_PATIENT, {patient: this.props.patient});
  }

  render() {
    const {patient} = this.props;

    return (
      <TouchableHighlight onPress={this.handlePress}>
        <View style={[containerStyles.container, listItemStyles.row]}>
          <Text style={listItemStyles.name}>{patient.name}</Text>
        </View>
      </TouchableHighlight>
    );
  }
}

// redux
const PatientListItem = connect(null, {push})(ListItem);

// wrapper
export default patient => (
  <PatientListItem patient={patient} />
);
