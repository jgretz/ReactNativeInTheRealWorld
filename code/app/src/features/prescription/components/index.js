export {default as PrescriptionListForPatient} from './prescriptionListForPatient';
