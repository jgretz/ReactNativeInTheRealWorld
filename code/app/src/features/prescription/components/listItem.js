import React from 'react';
import {View, Text} from 'react-native';

import containerStyles from '../../../styles/container';
import listItemStyles from '../styles/listItem';

// wrapper
export default script => (
  <View style={[containerStyles.container, listItemStyles.row]}>
    <Text style={listItemStyles.drug}>{script.drug}</Text>
  </View>
);
