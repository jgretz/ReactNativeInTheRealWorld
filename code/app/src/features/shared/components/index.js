export {default as Loading} from './loading';
export {default as LoginForm} from './loginForm';
export {default as LogoutButton} from './logoutButton';
