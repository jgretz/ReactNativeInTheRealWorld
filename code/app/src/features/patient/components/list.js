import React, {Component} from 'react';
import {connect} from 'react-redux';
import {View, ListView} from 'react-native';

import containerStyles from '../../../styles/container';
import listViewStyles from '../../../styles/listView';

import listItem from './listItem';
import {patientListViewDataSourceSelector} from '../selectors';

class List extends Component {
  static navigationOptions = {
    title: 'Patient List',
  };

  render() {
    return (
      <View style={containerStyles.container}>
        <ListView
          style={listViewStyles.list}
          dataSource={this.props.dataSource}
          renderRow={listItem}
          pageSize={22} />
      </View>
    );
  }
}

const mapStateToProps = state => ({
  dataSource: patientListViewDataSourceSelector(state),
});

export default connect(mapStateToProps)(List);
