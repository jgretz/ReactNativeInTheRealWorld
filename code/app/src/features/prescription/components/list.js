import React, {Component} from 'react';
import {View, ListView} from 'react-native';
import {connect} from 'react-redux';

import containerStyles from '../../../styles/container';
import listViewStyles from '../../../styles/listView';

import listItem from './listItem';
import {prescriptionsListViewDataSource} from '../selectors';

class List extends Component {
  static navigationOptions = ({navigation}) => ({
    title: `${navigation.state.params.patient.name}`,
  });

  render() {
    return (
      <View style={containerStyles.container}>
        <ListView
          style={listViewStyles.list}
          dataSource={this.props.dataSource}
          renderRow={listItem}
          pageSize={22}
          enableEmptySections />
      </View>
    );
  }
}

const mapStateToProps = (state, props) => ({
  dataSource: prescriptionsListViewDataSource(state, props),
});

export default connect(mapStateToProps)(List);
