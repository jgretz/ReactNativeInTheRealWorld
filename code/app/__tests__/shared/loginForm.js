/* eslint-disable */
import React, {Component} from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

import {LoginForm} from '../../src/features/shared/components/';

// create mockStore
const mockStore = configureStore();
const store = mockStore({});

// tests
describe('login', () => {
  let loginForm = null;
  beforeEach(() => {
    loginForm = renderer.create(
      <Provider store={store}>
        <LoginForm />
      </Provider>
    );
  });

  it('renders correctly', () => {
    expect(loginForm).toMatchSnapshot();
  });
});
