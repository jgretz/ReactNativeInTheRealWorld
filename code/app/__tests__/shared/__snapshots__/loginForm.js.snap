// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Sign In
  </Text>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 5,
          "marginTop": 15,
        }
      }
    >
      Email
    </Text>
    <TextInput
      autoCapitalize="none"
      name="email"
      onBlur={[Function]}
      onChangeText={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderWidth": 1,
          "fontSize": 16,
          "height": 40,
          "padding": 10,
          "width": 200,
        }
      }
      value=""
    />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 5,
          "marginTop": 15,
        }
      }
    >
      Password
    </Text>
    <TextInput
      name="password"
      onBlur={[Function]}
      onChangeText={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderWidth": 1,
          "fontSize": 16,
          "height": 40,
          "padding": 10,
          "width": 200,
        }
      }
      value=""
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "marginTop": 20,
          }
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
